sretoolbox/container/image.py

```diff
@@ -178,8 +219,13 @@ def manifest(self):
         Property to cache the manifest returned but get_manifest()
         """
         if self._cache_manifest is None:
-            manifest = self._get_manifest()
-            self._cache_manifest = manifest.json()
+            try:
+                manifest = self.digest_store[self.digest]
```

Comment:
There's a circular dependency here. `manifest()` calls `digest()` that calls `manifest()`. The cache should be ETag based, not digest based.

Reply:
Dang, I had missed the `or self.digest` part in `_fetch_manifest`. I'll break that dependency in a moment.

Reply:
... actually no. If you check the `test_parser` test, it calls `digest` on each of the images and we don't enter any kind of recursive loop or call cycle.

Reply:
great, tks for checking